src/main/java/rx/internal/operators/OperatorGroupBy.java

```diff
@@ -219,6 +219,13 @@ public void onCompleted() {
             if (done) {
                 return;
             }
+            List<GroupedUnicast<K, V>> list = new ArrayList<GroupedUnicast<K, V>>(groups.values());
+            groups.clear();
+            
+            for (GroupedUnicast<K, V> e : list) {
```

Comment:
The idea was to reduce the window when terminating groups remove themselves from the map; but I don't think such case is likely anymore. I'll update the PR accordingly.